cmd/thanos/compact.go

```diff
@@ -85,15 +85,16 @@ func registerCompact(app *extkingpin.App) {
 	conf := &compactConfig{}
 	conf.registerFlag(cmd)
 
-	cmd.Setup(func(g *run.Group, logger log.Logger, reg *prometheus.Registry, tracer opentracing.Tracer, _ <-chan struct{}, _ bool) error {
-		return runCompact(g, logger, tracer, reg, component.Compact, *conf, getFlagsMap(cmd.Flags()))
+	cmd.Setup(func(g *run.Group, logger log.Logger, reg *prometheus.Registry, tracer opentracing.Tracer, reqLogYAML []byte, _ <-chan struct{}, _ bool) error {
```

Comment:
hmmm, I think we should actually expose just abstracted config struct or someday even easier way to use those decision.

Maybe instead, we could pass extra middlwares, interceptors that implementation has to pass to servers

Reply:
We should create a TODO for features that we might want to add later someday :)

Reply:
Sorry but this is still not addressed. Which what I said  should be later?

I disagree that ` reqLogYAML []byte` should be passed. It does not make sense to parse yaml in each command. We do it in same why, so why replicating this code?

I would be ok with passing config here for now, but ideally middleware/internceptor would be ideal now. Should be easy. It's not future feature? We already pass logging interceptor/middlewars, not?

Reply:
Yeah, I decided to add flags to individual components, rather than passing a byte array among all the components, which seems quite error prone.